### PR TITLE
First implementation of local CRUD operations.

### PR DESCRIPTION
**Pull request by n1k0**
r=? @leplatrem (code style, quality of implementation, docs review, any missing bits?)

### CODE REVIEW

**Comment by n1k0** [comment]
This is now ready for review @leplatrem

File: src/collection.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        this._db = event.target.result;
+        resolve(this);
+      };
+    });
+  }
+
+  transaction(mode) {
+    return this._db
+      .transaction([this._collName], mode)
+      .objectStore(this._collName);
```
Comment (leplatrem) [inline src/collection.js]: This method is called transaction but actually starts a new transaction and returns an object store.

File: src/collection.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  get name() {
+    return this._collName;
+  }
+
+  init() {
+    return new Promise((resolve, reject) => {
+      var request = indexedDB.open(this._collName, 1);
+      request.onupgradeneeded = event => {
+        var store = event.target.result.createObjectStore(this._collName, {
```
Comment (leplatrem) [inline src/collection.js]: we could use ``this.name`` everywhere no ?

File: src/collection.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+      }
+      var request;
+      if (!record.id) {
+        request = transaction.add(Object.assign({}, record, {id: uuid4()}));
+      } else {
+        request = transaction.put(record);
+      }
+      request.onsuccess = function(event) {
+        resolve({
+          data: Object.assign({}, record, {id: event.target.result}),
```
Comment (leplatrem) [inline src/collection.js]: I don't understand this :( What are we providing in data exactly ?

File: src/index.js [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
+    this._collections = {};
   }
 
-  collection(name) {
-    return new Collection(name);
+  collection(collName) {
+    return new Promise((resolve, reject) => {
+      // if collection collName is missing, reject
+      if (!collName)
+        return reject(new Error("missing collection name"));
```
Comment (leplatrem) [inline src/index.js]: I wonder if this should faild in a promise, and if it should fail during call...

File: src/utils.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+export function attachFakeIDBSymbolsTo(obj) {
```
Comment (leplatrem) [inline src/utils.js]:
```
/**
 In FakeIndexedDB, symbols are exposed using ``FDB`` prefixes in names.
 This piece of code will register them with the same names as native API,
 only if indexedDB is not already available.
 */
```

**Comment by leplatrem** [comment]
r+ \o/

**Pull request merged**